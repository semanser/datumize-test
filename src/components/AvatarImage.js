import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ColorHash from 'color-hash'

class AvatarImage extends Component {
  constructor(props) {
    super(props)

    this.colorHash = new ColorHash()

    this.rolesColors = {
      Admin: '#EA522F',
      Editor: '#8245EA',
			Viewer: '#45EA69',
    }

    this.getShortName = this.getShortName.bind(this)
  }

  getShortName(name) {
    return name.split(' ')
      .map(n=>n[0])
      .join('')
  }

  render() {
    const style = this.props.type === 'role'
      ? {
        border: `2px solid ${this.rolesColors[this.props.name]}`,
        color: this.rolesColors[this.props.name],
      }
      : {
        background: `${this.colorHash.hex(this.props.name)}`,
      }

    const text = this.props.type === 'role'
      ? this.props.name[0]
      : this.getShortName(this.props.name)

    return (
      <div className="icon" style={style}>
        {text}
      </div>
    )
  }
}

AvatarImage.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
}

export default AvatarImage

