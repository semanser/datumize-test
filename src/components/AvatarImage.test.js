import React from 'react'
import ReactDOM from 'react-dom'
import ShallowRenderer from 'react-test-renderer/shallow'
import AvatarImage from './AvatarImage'

const renderer = new ShallowRenderer()

describe('AvatarImage Component', () => {
  it('Renders with type="name" without crashing', () => {
    const div = document.createElement('div')
    const props = {
      name: 'John Doe',
      type: 'name'
    }

    const tree = renderer.render(<AvatarImage {...props}/>)
    expect(tree).toMatchSnapshot()
  })

  it('Renders Admin role correctly', () => {
    const div = document.createElement('div')
    const props = {
      name: 'Admin',
      type: 'role'
    }

    const tree = renderer.render(<AvatarImage {...props}/>)
    expect(tree).toMatchSnapshot()
  })

  it('Renders Editor role correctly', () => {
    const div = document.createElement('div')
    const props = {
      name: 'Editor',
      type: 'role'
    }

    const tree = renderer.render(<AvatarImage {...props}/>)
    expect(tree).toMatchSnapshot()
  })

  it('Renders Viewer role correctly', () => {
    const div = document.createElement('div')
    const props = {
      name: 'Viewer',
      type: 'role'
    }

    const tree = renderer.render(<AvatarImage {...props}/>)
    expect(tree).toMatchSnapshot()
  })
})

