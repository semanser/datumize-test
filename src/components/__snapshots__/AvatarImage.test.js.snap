// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarImage Component Renders Admin role correctly 1`] = `
<div
  className="icon"
  style={
    Object {
      "border": "2px solid #EA522F",
      "color": "#EA522F",
    }
  }
>
  A
</div>
`;

exports[`AvatarImage Component Renders Editor role correctly 1`] = `
<div
  className="icon"
  style={
    Object {
      "border": "2px solid #8245EA",
      "color": "#8245EA",
    }
  }
>
  E
</div>
`;

exports[`AvatarImage Component Renders Viewer role correctly 1`] = `
<div
  className="icon"
  style={
    Object {
      "border": "2px solid #45EA69",
      "color": "#45EA69",
    }
  }
>
  V
</div>
`;

exports[`AvatarImage Component Renders with type="name" without crashing 1`] = `
<div
  className="icon"
  style={
    Object {
      "background": "#3c3a78",
    }
  }
>
  JD
</div>
`;
